test-framework/sudo-compliance-tests/src/visudo/kill-visudo.sh

```diff
@@ -0,0 +1,6 @@
+visudopid=$(pidof visudo | sort -gr | cut -f 1 -d ' ')
+
+if [ -n "$visudopid" ]; then
+    sleep 1
```

Comment:
I'd suggest using a "barrier file" to synchronize this with the editor script. have the editor script `touch /tmp/barrier` before sleeping and have this kill script busy wait until that file is created, e.g. `until [ -f /tmp/barrier ]; do sleep 0.1; done`